src/components/networked-video-source.js

```diff
@@ -0,0 +1,87 @@
+/* global AFRAME, NAF, THREE */
+var naf = require('../NafIndex');
+
+AFRAME.registerComponent('networked-video-source', {
+
+  schema: {
+  },
+
+  dependencies: ['material'],
+
+  init: function () {
+    this.videoTexture = null;
+    this.video = null;
+    this.stream = null;
+
+    this._setMediaStream = this._setMediaStream.bind(this);
+
+    NAF.utils.getNetworkedEntity(this.el).then((networkedEl) => {
+      const ownerId = networkedEl.components.networked.data.owner;
+
+      if (ownerId) {
+        NAF.connection.adapter.getMediaStream(ownerId, "video")
+          .then(this._setMediaStream)
+          .catch((e) => naf.log.error(`Error getting media stream for ${ownerId}`, e));
+      } else {
+        // Correctly configured local entity, perhaps do something here for enabling debug audio loopback
+      }
+    });
+  },
+
+  _setMediaStream(newStream) {
+
+    if(!this.video) {
+      this.setupVideo();
+    }
+
+    if(newStream != this.stream) {
+      if (this.stream) {
+        this._clearMediaStream();
+      }
+
+      if (newStream) {
+        this.video.srcObject = newStream;
+
+        var playResult = this.video.play();
+        if (playResult instanceof Promise) {
+          playResult.catch((e) => naf.log.error(`Error play video stream`, e));
+        }
+
+        const mesh = this.el.getObject3D('mesh');
+        mesh.material.map = new THREE.VideoTexture(this.video);
```

Comment:
Fo you want to set `this.videoTexture` here?

Reply:
yeah was not sure, let do that.